src/settings.js

```diff
@@ -1753,6 +1753,14 @@ var DEFAULT_TO_CXX = 1;
 // long double printing precision.
 var PRINTF_LONG_DOUBLE = 0;
 
+// Run wabt's wasm2c tool on the final wasm, and combine that with a C runtime,
+// resulting in a .c file that you can compile with a C compiler to get a
+// native executable that works the same as the normal js+wasm. This will also
+// emit the wasm2c .h file. The output filenames will be X.wasm.c, X.wasm.h
+// if your output is X.js or X.wasm (note the added .wasm. we make sure to emit,
+// which avoids trampling a C file).
+var WASM2C = 0;
```

Comment:
I agree it would be nice to be more modular. But I don't think it would be practical in this case, since the code needs to look at the compiler state - right now it already looks at `AUTODEBUG` and `EXPECT_MAIN`, and it will need more for things like FS support options. In theory we could ask the user to pass the flags twice (to emcc and to the post-link tool) but I think that's not as usable.

I think this is a non-intrusive change, at least - this won't thread itself around in emcc.py. It's just one chunk of code in shared.py.